### DISCUSSION TITLE
What features should we build next?

### DISCUSSION THREAD

**Discussion by tabacitu** [Laravel-Backpack/community-forum#751] [Announcements]
It's Q4 2023. We've already finished or started working on everything you guys have asked for last time. So our schedule is clear for 2024, we can add stuff to it 🎉

So let's do this again. **Let's vote on what features YOU need most, so we can prioritize and build them** 💪

Please only vote if you think you will need it 🙏 Voting on everything will just make us waste our resources on features that don't actually matter. If you have things you absolutely think we should build and aren't mentioned here, please let us know.

Let's do this 💪

**Comment by tabacitu** [comment]
support for Tailwind CSS

**Comment by tabacitu** [comment]
edit related entries in Show Operation (aka. CRUD inside CRUD)

**Comment by tabacitu** [comment]
DevTools to allow editing models, migrations etc. (not just creating)

**Comment by tabacitu** [comment]
Livewire components (table, form, widgets etc.)

**Reply by munjaldevelopment** [reply]
Eagerly waiting for the same..

**Comment by tabacitu** [comment]
HTMX components (table, form, widgets etc.)

**Comment by tabacitu** [comment]
Interface for the admin to run Artisan commands

**Reply by Astriel** [reply]
This could be usefull to avoid logging to putty or SSH on the server. Has to be compatible with the permission system to avoid all users to execute Artisan commands.

**Reply by ziming** [reply]
could be a security risk tho

**Comment by tabacitu** [comment]
interface for the admin to easily translate lang strings to other languages
(like LangFileManager but that doesn't override PHP files)

**Reply by karandatwani92** [reply]
https://github.com/Laravel-Backpack/translation-manager

**Comment by tabacitu** [comment]
Calendar Operation - alternative to ListOperation that shows entries on a FullCalendar

**Reply by karandatwani92** [reply]
Done! Available here:
https://backpackforlaravel.com/products/calendar-operation

**Reply by ziming** [reply]
Looks really great!

**Comment by tabacitu** [comment]
KanBan Operation - alternative to ListOperation that shows entries in multiple columns on a KanBan board

**Reply by munjaldevelopment** [reply]
After "Calendar" operation, Kanban should be next in "todo basket list"

**Comment by tabacitu** [comment]
Grid Operation - alternative to ListOperation that show entries in a grid format (eg. a contacts list)

**Comment by tabacitu** [comment]
drag & drop page builder with blocks, so the admin can create, edit and customize presentation pages more deeply (eg. GrapeJS or EditorJS)

**Comment by tabacitu** [comment]
Export Operation - ability to export tables containing millions of rows

**Reply by Astriel** [reply]
This could be really usefull for big production websites. Will be needed by a lot of people over time.

**Comment by tabacitu** [comment]
more themes

**Reply by exorgroup** [reply]
would be nice to have metronic theme

**Reply by munjaldevelopment** [reply]
Metronic is awesome. Would be great if we use this theme. But I think there may be copyright issue with authors

**Reply by exorgroup** [reply]
All you need is a discussion with them - to send you the files they give you a download code - it could be a way to check that you paid a licence

**Reply by exorgroup** [reply]
AdminLTE is somewhat similar to Metronic and have a free option - so that is another option

**Comment by tabacitu** [comment]
Ability to work with APIs out of the box (not just Eloquent Models)

**Reply by Astriel** [reply]
That could be great to have a way to communicate either with external APIs to import data on Backpack for example, but also sending data to the API. Could be the Laravel internal API or and external API with Guzzle ?

**Comment by tabacitu** [comment]
complete removal of jQuery (still could be used, but optional)

**Comment by sjefmoling** [comment]
listviews of large/complex models combined with exports to excel/pdf to display overviews (not CRUD).

**Comment by sjefmoling** [comment]
Authorization on operation level based on policies/gates

**Reply by tabacitu** [reply]
You can actually already do that, it's not that difficult! I'll ask @karandatwani92 or @phpfour to write an article about it 😉

**Reply by karandatwani92** [reply]
Hey @pxpm, A gentle reminder for the article 😜

**Comment by sjefmoling** [comment]
wizard style forms

**Comment by pxpm** [comment]
1-Auth & Social Logins

**Comment by exorgroup** [comment]
Table Group by Column

basically being able to group the table based on a column value and create like a drop down on the group that when opened it will display all records

To give example - table with employees can have a group on Country and Locality 

Country: UK
* Locality: London
  * emp 1
  * emp 2
* Locality: Liverpool
  * emp 1
  * emp 2

Country: Italy
* Locality: Milan
  * emp 1
  * emp 2
* Locality: Rome
  * emp 1
  * emp 2

**Comment by exorgroup** [comment]
Dual Multi List 

this will have 2 lists normally with a set of buttons in between to transfer the selected items from the left list to the right list and vice versa - buttons would normally be <<, <, >, >>

**Reply by pxpm** [reply]
Something like this https://backpackforlaravel.com/docs/6.x/crud-fields#select_and_order-pro ? It's drag'n drop tho, no buttons.

**Reply by exorgroup** [reply]
Very similar concept so easy to implement in next version :)  

Main difference is that does one at a time whilst what I am suggesting does a number at a time if needed

**Comment by exorgroup** [comment]
QR code and Barcode fields - could be very useful - pass them a field value and they generate the desired image

**Reply by karandatwani92** [reply]
It is easy using https://github.com/milon/barcode

**Comment by exorgroup** [comment]
Switch List - similar to checkbox list but will have switches instead of checkbox

**Comment by exorgroup** [comment]
Timeline - display a list of data in a timeline fashion - could be used in projects where communication is involved like feedback, chat, email thread etc

**Comment by exorgroup** [comment]
Map - could be useful in project that have logistics, tracking and real estate

**Comment by prodixx** [comment]
use of backpack CRUD forms outside of admin area, into the public area

**Comment by prodixx** [comment]
InlineUpdate Operation - if you want to update an related entry on the fly (in a modal)

**Comment by pxpm** [comment]
I think a very cool package we can build is a Backpack-Sushi (https://github.com/calebporzio/sushi) integration, so that you can have a `ListOperation` that can show any arbitrary information like files in a folder (like we do in Devtools), arrays of data, api results etc. 
That way you can keep using all familiar crud functions in those tables like buttons, wrappers with href, filters, search, pagination etc.

**Comment by Astriel** [comment]
This really needs to be implemented as a plugin / core component of backpack : https://github.com/izica/relations-widgets-for-backpack/
This is a real struggle at the moment to implement sublist / relationships or tables on setupShowOperation. This was planned in th V6, but postponed. Would be perfect if it could work with the tab system.

**Reply by fronbow** [reply]
I agree, I found that component a while ago and use my own hacked version that does things slightly different for my admin panel!!

**Reply by Astriel** [reply]
I'm using ChatGPT to help me building custom components for the setupShowOperation but this is kinda complex and taking time. I would love to know more about your "hacked" version. I tried to install this one, but not working with V6.

**Reply by fronbow** [reply]
I just emailed you matey :)

**Comment by fronbow** [comment]
I would love a v6 version of [https://github.com/novius/laravel-backpack-crud-extended?tab=readme-ov-file#crud-boxes](https://github.com/novius/laravel-backpack-crud-extended?tab=readme-ov-file#crud-boxes) as an alternative to tabs for laying out my forms. It would make complex forms easier to manage both for me as the developer and my users!

**Reply by Astriel** [reply]
+1 this is a good idea.

**Comment by lieto-dk** [comment]
Move editable columns into the pro product !

**Comment by lieto-dk** [comment]
Have a select option in table columns.

**Comment by eleven59** [comment]
An easy way to customize (error/success) messages when overriding CRUD functions. For example, preventing deletion of specific IDs, and showing why it didn’t work instead of “may not have been deleted”

**Comment by ziming** [comment]
A backpack comments package (maybe use spatie laravel comments package behind the scenes)

**Comment by karandatwani92** [comment]
Buttons API to move a few items to the dropdown menu while keeping a few others inline. Discussed [here](https://github.com/Laravel-Backpack/community-forum/discussions/973)

**Comment by karandatwani92** [comment]
Scheduling Task Manager

**Comment by ericjanofski** [comment]
**Repeatable Field Group Templates**
I would love to have the ability to create field group templates for the repeatable field, similar to ACF Flexible Content.

Example: if our content team needs a field to hold either an image with a caption, a video with a thumnbail and caption or a section of text with a header and some body text, I currently need to build it all out with conditional javascript. If we could create json templates within the repeatable field and still have access to javascript and request verification, that would help so much!

There was a work in progress for v4 that had this functionality. I used it on a project for a large-ish project and it worked like a dream. Perhaps that could be revived? Thank you very much.

**Reply by tabacitu** [reply]
Ugh. This is a big one. I did love the previous "block editor" too, but the JS library wasn't maintained at all. Thanks for the suggestion Eric, will consider. Let's see if more people think it's worth building.
